lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+      if err.message =~ /80070057/m
+        trigger = nil
+      else
+        raise
+      end
+    end
+    
+    trigger
+  end
+
+  # DONE
+  def append_trigger(trigger_hash)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    obj = definition.Triggers.create(trigger_hash['type'])
+
+    set_properties_from_hash(obj, trigger_hash)
+
+    obj
+  end
+
+  # DONE
+  def set_trigger(index, trigger_hash)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    obj = definition.Triggers.Item(index)
+    
+    set_properties_from_hash(obj, trigger_hash)
+
+    obj
+  end
+
+  # Returns the status of the currently active task. Possible values are
+  # 'ready', 'running', 'queued', 'disabled' or 'unknown'.
+  #
+  # DONE
+  def status
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    case @pITask.State
+      when TASK_STATE_READY
+          status = 'ready'
+      when TASK_STATE_RUNNING
+          status = 'running'
+      when TASK_STATE_QUEUED
+          status = 'queued'
+        when TASK_STATE_DISABLED
+          status = 'disabled'
+        else
+          status = 'unknown'
+    end
+
+    status
+  end
+
+  # Returns the exit code from the last scheduled run.
+  #
+  # DONE
+  def exit_code
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    # Note Exit Code 267011 is generated when the task has never been run
+    status = @pITask.LastTaskResult
+
+    status
+  end
+
+  # Returns the comment associated with the task, if any.
+  #
+  # DONE
+  def comment
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.RegistrationInfo.Description
+  end
+
+  # Sets the comment for the task.
+  #
+  # DONE
+  def comment=(comment)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Description = comment
+
+    comment
+  end
+
+  # Returns the name of the user who created the task.
+  #
+  # DONE
+  def creator
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Author
+  end
+
+  # Sets the creator for the task.
+  #
+  # DONE
+  def creator=(creator)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Author = creator
+
+    creator
+  end
+
+  # Returns a Time object that indicates the next time the task will run.
+  # nil if the task has no scheduled time
+  #
+  # DONE
+  def next_run_time
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    time = @pITask.NextRunTime
+
+    # The API will still return a time WAAAY in the past if there is no schedule.
+    # As this is looking forward, if the next execution is 'scheduled' in the 1900s assume
```

Comment:
I wonder if this is the minimum OLE datetime value? https://msdn.microsoft.com/en-us/library/system.datetime.tooadate(v=vs.110).aspx

```
An OLE Automation date is implemented as a floating-point number whose integral component is the number of days before or after midnight, 30 December 1899, and whose fractional component represents the time on that day divided by 24. For example, midnight, 31 December 1899 is represented by 1.0; 6 A.M., 1 January 1900 is represented by 2.25; midnight, 29 December 1899 is represented by -1.0; and 6 A.M., 29 December 1899 is represented by -1.25.
The base OLE Automation Date is midnight, 30 December 1899. The minimum OLE Automation date is midnight, 1 January 0100. The maximum OLE Automation Date is the same as DateTime.MaxValue, the last moment of 31 December 9999.
```